stable/kong/templates/deployment.yaml

```diff
@@ -76,12 +76,20 @@ spec:
         - name: KONG_ADMIN_LISTEN
           value: 0.0.0.0:{{ .Values.admin.containerPort }}
         {{- end }}
-        {{- if .Values.proxy.useTLS }}
+        {{- if and .Values.proxy.http.enabled .Values.proxy.tls.enabled }}
         - name: KONG_PROXY_LISTEN
-          value: "0.0.0.0:{{ .Values.proxy.containerPort }} ssl"
-        {{- else }}
+          value: "0.0.0.0:{{ .Values.proxy.http.containerPort }},0.0.0.0:{{ .Values.proxy.tls.containerPort }} ssl"
+        {{ else }}
+
+        {{- if .Values.proxy.http.enabled }}
+        - name: KONG_PROXY_LISTEN
+          value: "0.0.0.0:{{ .Values.proxy.http.containerPort }}"
+        {{- end }}
+        {{- if .Values.proxy.tls.enabled }}
         - name: KONG_PROXY_LISTEN
-          value: 0.0.0.0:{{ .Values.proxy.containerPort }}
+          value: "0.0.0.0:{{ .Values.proxy.tls.containerPort }} ssl"
+        {{- end }}
```

Comment:
Can we simplify this logic?
Rather than using so many `if` blocks, can we construct a string of HTTP and TLS listeners and assign it once only?
You can use the `templates/_helpers.tpl` as well.

Reply:
Good idea.  Will update.